nuxeo-services/nuxeo-platform-usermanager-api/src/main/java/org/nuxeo/ecm/platform/usermanager/UserManager.java

```diff
@@ -58,6 +58,12 @@
      */
     NuxeoPrincipal getPrincipal(String username);
```

Comment:
Could be interesting to make this method `default` and provide implementation you give to `UserManagerImpl`.